### PR TITLE
Convert remaining San d'Oria Missions to Interaction Framework

### PR DESCRIPTION
**Pull request by claywar**
**_I affirm:_**
- [x] I have paid attention to this example and will edit again if need be to not break the formatting, or I will be ignored
- [x] that I agree to LandSandBoat's [Limited Contributor License Agreement](https://github.com/LandSandBoat/server/blob/base/.github/CONTRIBUTOR_AGREEMENT.md), as written on this date
- [x] that I have **read the [Contributing Guide](https://github.com/LandSandBoat/server/blob/base/CONTRIBUTING.md)**
- [x] that I've _**tested my code and things my code changed**_ since the last commit in the PR, and will test after any later commits

Converts remaining San d'Oria missions from 5-2 through 9-2 to Interaction Framework.

### CODE REVIEW

**Comment by Xaver-DaRed** [comment]
Haven't seen anything weird, looks very good. Gonna run 2 new characters through the whole sandorian mission line with a friendo to test it.

**Comment by claywar** [comment]
Just completed a fairly exhaustive run from beginning to end; fixed issues encountered.

File: scripts/globals/missions.lua [hunk 1]
```diff
@@ -943,28 +937,7 @@ function finishMissionTimeline(player, guard, csid, option)
...
-                19,            {1044, 0},                  {1043, 0},                  {0, 0},                   {0, 0},                   {{14, 0}, {6}, {3, "SecretWeaponStatus", 0}, {9, 284}, {11, 8}, {8, 60000}, {12}}, -- MISSION 7-2 (Finish)
-                20,               {0, 0},                     {0, 0},                {102, 0},                   {0, 0},                   {{14, 0}, {9, 288}, {5, 800}, {12}},                                    -- MISSION 8-1 (Finish)
-                21,               {0, 0},                     {0, 0},                {104, 0},                   {0, 0},                   {{14, 0}, {9, 284}, {11, 9}, {8, 80000}, {6}, {12}},                    -- MISSION 8-2 (Finish (Door: Great Hall))
-                22,               {0, 0},                     {0, 0},                 {76, 0},                   {0, 0},                   {{14, 0}, {9, 481}, {9, 482}, {9, 483}, {5, 900}, {12}}                 -- MISSION 9-1 (Finish (Door: Great Hall))
-            }
-        end
-    elseif (nation == xi.nation.BASTOK) then
```
Comment (zach2good) [inline scripts/globals/missions.lua:967]: 🙏

File: scripts/missions/sandoria/5_2_The_Shadow_Lord.lua [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+                        player:getMissionStatus(mission.areaId) == 3 and
+                        player:getLocalVar("battlefieldWin") == 160
+                    then
+                        if
+                            player:getCurrentMission(xi.mission.log_id.ZILART) ~= xi.mission.id.zilart.THE_NEW_FRONTIER and
+                            not player:hasCompletedMission(xi.mission.log_id.ZILART, xi.mission.id.zilart.THE_NEW_FRONTIER)
+                        then
+                            -- Don't add missions we already completed. Players who change nation will hit this.
+                            player:addMission(xi.mission.log_id.ZILART, xi.mission.id.zilart.THE_NEW_FRONTIER)
+                        end
```
Comment (zach2good) [inline scripts/missions/sandoria/5_2_The_Shadow_Lord.lua:175]: Smart

File: scripts/missions/sandoria/7_2_The_Secret_Weapon.lua [hunk 3]
```diff
@@ -0,0 +1,211 @@
...
+
+local handleAcceptMission = function(player, csid, option, npc)
+    if option == 19 then
+        mission:begin(player)
+    end
+end
+
+-- This mission contains up to 3 statuses that are used before the mission is actually accepted, and is blocking
+-- for other missions when the requirements are met.  Until those requirements are met, gate guards will default
+-- to their typical mission assignment behavior.
```
Comment (zach2good) [inline scripts/missions/sandoria/7_2_The_Secret_Weapon.lua:35]: 👍

File: scripts/missions/sandoria/9_1_Breaking_Barriers.lua [hunk 4]
```diff
@@ -0,0 +1,224 @@
...
+    rankPoints = 900,
+}
+
+local handleAcceptMission = function(player, csid, option, npc)
+    if option == 22 then
+        mission:begin(player)
+
+        -- At this point, Prince cutscenes from M8-2 are no longer available.  Reset this var for
+        -- cleanliness.
+        player:setCharVar('Mission[0][21]Option', 0)
```
Comment (zach2good) [inline scripts/missions/sandoria/9_1_Breaking_Barriers.lua:38]: 👍

File: scripts/missions/sandoria/9_2_The_Heir_to_the_Light.lua [hunk 5]
```diff
@@ -0,0 +1,401 @@
...
+                -- should inventory be full.  Set 'Flag' mission variable should this occur.
+                -- 'Option' set after complete for additional dialogue, and cleared on Zoning
+                -- into Southern San d'Oria.
+                [9] = function(player, csid, option, npc)
+                    mission:complete(player)
+                    mission:setVar(player, 'Option', 1)
+
+                    if not npcUtil.giveItem(player, 181) then
+                        mission:setVar(player, 'Flag', 1)
+                    end
```
Comment (zach2good) [inline scripts/missions/sandoria/9_2_The_Heir_to_the_Light.lua:219]: 👍

File: scripts/zones/Batallia_Downs/DefaultActions.lua [hunk 6]
```diff
@@ -0,0 +1,5 @@
+local ID = require("scripts/zones/Batallia_Downs/IDs")
+
+return {
+    ['qm3']     = { messageSpecial = ID.text.NOTHING_OUT_OF_ORDINARY },
+}
```
Comment (zach2good) [inline scripts/zones/Batallia_Downs/DefaultActions.lua]: Funky formatting in here

File: scripts/zones/Horlais_Peak/DefaultActions.lua [hunk 7]
```diff
@@ -0,0 +1,5 @@
+local ID = require("scripts/zones/Horlais_Peak/IDs")
+
+return {
+    ['Hot_Springs']     = { messageSpecial = ID.text.NOTHING_OUT_OF_ORDINARY },
+}
```
Comment (zach2good) [inline scripts/zones/Horlais_Peak/DefaultActions.lua]: Formatting is a bit wonky in here

**Review by zach2good — APPROVED** [review]
Sandy NPCs looking _fresh_.

I'd say take a look at the formatting in the `DefaultActions.lua` files, otherwise, it's all looking spick and span 👍

**Pull request merged**